cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaRoller.java

```diff
@@ -826,44 +925,66 @@ protected Admin adminClient(Set<NodeRef> nodes, boolean ceShouldBeFatal) throws
         int controller = controller(nodeRef, operationTimeoutMs, TimeUnit.MILLISECONDS, restartContext);
-        int stillRunning = podToContext.reduceValuesToInt(100, v -> v.promise.future().isComplete() ? 0 : 1,
-                0, Integer::sum);
-        return controller == nodeRef.nodeId() && stillRunning > 1;
+        if (controller == nodeRef.nodeId()) {
+            int stillRunning = podToContext.reduceValuesToInt(100, v -> v.promise.future().isComplete() ? 0 : 1,
+                    0, Integer::sum);
+            return stillRunning > 1;
+        } else {
+            return false;
+        }
     }
 
     /**
-     * Completes the returned future <strong>on the context thread</strong> with the id of the controller of the cluster.
+     * Completes the returned future <strong>on the context thread</strong> with the id of the controller of the cluster
+     * or the active controller when running in KRaft mode.
      * This will be -1 if there is not currently a controller.
      *
      * @return A future which completes the node id of the controller of the cluster, or -1 if there is not currently a controller.
      */
     @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE") // seems to be completely spurious
     int controller(NodeRef nodeRef, long timeout, TimeUnit unit, RestartContext restartContext) throws Exception {
-        // Don't use all allClient here, because it will have cache metadata about which is the controller.
-        try (Admin ac = adminClient(Set.of(nodeRef), false)) {
-            Node controllerNode = null;
-            try {
-                DescribeClusterResult describeClusterResult = ac.describeCluster();
-                KafkaFuture<Node> controller = describeClusterResult.controller();
-                controllerNode = controller.get(timeout, unit);
-                restartContext.clearConnectionError();
-            } catch (ExecutionException | TimeoutException e) {
-                maybeTcpProbe(nodeRef, e, restartContext);
+        int id;
+        if (nodeRef.controller()) {
+            id = await(restartContext.quorumCheck.quorumLeaderId(), timeout, unit,
+                    t -> new UnforceableProblem("An error while trying to determine the quorum leader id", t));
+            LOGGER.debugCr(reconciliation, "KRaft active controller is {}", id);
+        } else {
+            // TODO Either this is a KRaft broker or ZooKeeper broker. Since KafkaRoller does not know if this is KRaft mode or
+            //      not continue with describeCluster. In KRaft mode this returns a random broker and will mean this broker is deferred.
+            //      In future this can be improved by telling KafkaRoller whether the cluster is in KRaft mode or not.
+            //      This is tracked in https://github.com/strimzi/strimzi-kafka-operator/issues/9373.
+            // Don't use broker admin client here, because it will have cache metadata about which is the controller.
```

Comment:
This is not correct. If you look at the code for `KafkaAdmin#describeCluster()`, for example, you'll see that it _always_ sends a `DescribeCluster` request (or a `Metadata` request), so the results we get from it never come from a cache _within the client_:

* Those results come from a (random) broker, which serves the response from a cache, and therefore the results can be stale, but the staleness is bounded. (And the possibility of staleness is intrinsic to working with distributed systems).
* The Admin client does have its own metadata cache, which it uses for determining which broker to send requests to (when request must go to a specific broker), but it's not used for generating results to Admin operations. 

There might be valid reasons to spinning up yet another admin client here, but I don't believe admin client caching is one.

Reply:
@tombentley just to be sure what you are asking for here ... it's that just the comment is wrong because caching doesn't happen in the client but on the broker. It's not clear to me if you are asking for some other change.

Reply:
Should we then use the admin client that has been initialised with all of the brokers, rather than creating another admin against a specific broker?

Reply:
Also just to clarify, this logic was here before this PR. It was moved due to the refactor we have done. If we don't want to use a new admin client here, should this perhaps be changed in a separate PR?